src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -1220,6 +1222,30 @@ public static void Reverse<T>(this Span<T> span)
                 return;
             }
 
+            if (typeof(T).IsValueType)
```

Comment:
```suggestion
            if (!RuntimeHelpers.IsReferenceOrContainsReferences<T>())
```

Reply:
You're right, much better! I'll make the change.